hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/request/key/TestOMKeyDeleteRequestWithFSO.java

```diff
@@ -227,4 +236,63 @@ public void testRecursiveAccessCheck() throws Exception {
     // false for file1.
     assertFalse(pathViewer.isCheckRecursiveAccess());
   }
+
+  @Test
+  public void testDeleteDirectoryWithColonInFSOBucket() throws Exception {
+
+    when(ozoneManager.getEnableFileSystemPaths()).thenReturn(true);
+
+    OMRequestTestUtils.addVolumeAndBucketToDB(volumeName, bucketName, omMetadataManager, getBucketLayout());
+
+    String dirName = "foo:dir/";
+
+    String dirKeyPath = addKeyToDirTable(volumeName, bucketName, dirName);
+
+    long parentObjectID = 0L;
+    long dirObjectID = 12345L;
+
+    OmDirectoryInfo omDirectoryInfo = OMRequestTestUtils.createOmDirectoryInfo(dirName, dirObjectID, parentObjectID);
+
+    omMetadataManager.getDirectoryTable().put(dirKeyPath, omDirectoryInfo);
+
+    OmDirectoryInfo storedDirInfo = omMetadataManager.getDirectoryTable().get(dirKeyPath);
+    assertNotNull(storedDirInfo);
+    assertEquals(dirName, storedDirInfo.getName());
+    assertEquals(dirObjectID, storedDirInfo.getObjectID());
+    assertEquals(parentObjectID, storedDirInfo.getParentObjectID());
+
+    OMRequest deleteRequest = doPreExecuteCheck(createDeleteKeyRequest(dirName));
+
+    OMKeyDeleteRequest omKeyDeleteRequest = new OMKeyDeleteRequestWithFSO(deleteRequest, getBucketLayout());
```

Comment:
Use `getOmKeyDeleteRequest` instead of `new OMKeyDeleteRequestWithFSO`.  It creates different type for `TestOMKeyDeleteRequest` and `TestOMKeyDeleteRequestWithFSO`.